userspace/libscap/engine/bpf/scap_bpf.c

```diff
@@ -528,10 +528,18 @@ static int32_t load_tracepoint(struct bpf_engine* handle, const char *event, str
 	fd = bpf_load_program(prog, program_type, insns_cnt, error, BPF_LOG_SIZE, prog_name);
 	if(fd < 0)
 	{
-		fprintf(stderr, "-- BEGIN PROG LOAD LOG --\n%s\n-- END PROG LOAD LOG --\n", error);
-		snprintf(handle->m_lasterr, SCAP_LASTERR_SIZE, "libscap: bpf_load_program() err=%d event=%s", errno, event);
-		free(error);
-		return SCAP_FAILURE;
+		/* It is possible than some old kernels don't support the prog_name so in case
+		 * of loading failure, we try again the loading without the name. See it in libbpf:
+		 * https://github.com/torvalds/linux/blob/master/tools/lib/bpf/libbpf.c#L4926
+		 */
+		fd = bpf_load_program(prog, program_type, insns_cnt, error, BPF_LOG_SIZE, NULL);
```

Comment:
quick question: isn't there any indication of the type of error that happened? Like an error code or something similar? trying twice regardless seems a bit blunt, but it may be required especially in case of bugs in specific versions

Reply:
It returns E2BIG on our testing machine; but given that we are not sure that the `errno` is set to the same error on every kernel/arch, we decided to be a little more safe here!

Note that E2BIG makes no sense in that situation too :D

Reply:
E2BIG normally means something else (too many instructions in the bpf program or perhaps some verifier failures due to back edges that would create large error buffers but I'm not sure) so I agree. And I imagine that adopting the approach linked in libbpf (check if some feature exists) cannot be done reliably here, right?

Reply:
Yeah, `e2big` is really strange in this case, for this reason, a massive retry is better IMO. For what concern `libbpf` check, under the hood it performs almost the same thing, it tries to inject a program and check if the feature is here or not, obviously in that case it is quite elegant but we cannot copy all their stuff for just this check